analytics.js

```diff
@@ -196,13 +196,35 @@ class Analytics {
         );
       });
 
+      let staging_env = false; // by default staging env is set to false
+
+      // Get the CDN base URL from the script tag
+      const scripts = document.getElementsByTagName("script");
+      for (let i = 0; i < scripts.length; i += 1) {
+        const curScriptSrc = removeTrailingSlashes(
+          scripts[i].getAttribute("src")
+        );
+        // only in case of staging SDK staging env will be set to true
+        if (
+          curScriptSrc &&
+          curScriptSrc.startsWith("http") &&
+          curScriptSrc.endsWith("rudder-analytics-staging.min.js")
+        ) {
+          staging_env = true;
+        }
+      }
```

Comment:
Why repeat this again? Let's set the flag when we first check

Reply:
Thought of doing so but if destSDKBaseURL id provided then it wouldn’t go to that code block

Reply:
Makes sense. Let's re-factor this into a function.
The last thing we want to do is add redundant code.

Reply:
done